src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -88,14 +88,23 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     }
 
     public async getCredentials(): Promise<AWS.Credentials> {
-        const validationMessage = this.validate()
-        if (validationMessage) {
-            throw new Error(`Profile ${this.profileName} is not a valid Credential Profile: ${validationMessage}`)
-        }
+        const originalStsConfiguration = AWS.config.sts
```

Comment:
Can this be done not this way? and if it needs to be done this way, this is not thread safe

Reply:
It is gross. I'm waiting on https://github.com/aws/aws-sdk-js/issues/3088 to see if there is better guidance.

Reply:
In this case, can we use `AsyncLock` or is that even grosser?